cpp/node_similarity_module/algorithms/node_similarity.hpp

```diff
@@ -0,0 +1,206 @@
+#include <mgp.hpp>
+#include <math.h>
+
+
+namespace node_similarity_util {
+
+struct pair_hash {
+    template <class T1, class T2>
+    std::size_t operator () (std::pair<T1, T2> const &pair) const
+    {
+        std::size_t h1 = std::hash<T1>()(pair.first);
+        std::size_t h2 = std::hash<T2>()(pair.second);
+
+        return h1 ^ h2;
+    }
+};
+
+enum Similarity { jaccard, overlap, cosine };
+
+// Methods
+constexpr char const *jaccardAll = "jaccard";
+constexpr char const *jaccardPairwise = "jaccard_pairwise";
+constexpr char const *overlapAll = "overlap";
+constexpr char const *overlapPairwise = "overlap_pairwise";
+constexpr char const *cosineAll = "cosine";
+constexpr char const *cosinePairwise = "cosine_pairwise";
+// Parameter object names
+constexpr char const *src_nodes = "src_nodes";
+constexpr char const *dst_nodes = "dst_nodes";
+constexpr char const *prop_vector = "property";
+// Return object names
+char const *node1_name = "node1";
+char const *node2_name = "node2";
+char const *similarity = "similarity";
```

Comment:
In other modules like pagerank and so on, we set this constants in top module, here `node_similarity_module`. Also, constants are usually file based if I am not mistaken, here you can check this [comment](https://stackoverflow.com/questions/12042549/define-constant-variables-in-c-header#:~:text=30,17%20inline%20variables). I see that you don't need them here, and that is good, so move them to top module

Reply:
Addressed